### PR TITLE
Type inference tweaks due to solc-typed-ast changes

### PR DESCRIPTION
**Pull request by blitz-1306**
## Status
This PR opened to check how [type inference changes](https://github.com/ConsenSys/solc-typed-ast/pull/142) in **solc-typed-ast** will affect Scribble. **Do not consider merging, please**.

## Changes
- [x] Use WIP type inference logic from **solc-typed-ast**.

## Notes
The local version of **solc-typed-ast** is used as the dependency for now. CI would be broken for a while.

Regards.

### CODE REVIEW

**Review by cd1m0 — APPROVED** [review]
LGTM. However lets not merge before we do the followup fixes in solc-typed-ast

**Comment by blitz-1306** [comment]
Merging this and doing a release would also fix #195.

**Review by cd1m0 — APPROVED** [review]
LGTM to merge after tests complete

**Pull request merged**
